tests/test_batch_run.py

```diff
@@ -83,8 +83,8 @@ def get_local_model_param(self):
         return 42
 
     def step(self):
-        self.datacollector.collect(self)
         self.schedule.step()
+        self.datacollector.collect(self)
```

Comment:
Why is this moved after the schedule step?

Reply:
Great question. That's because this way data is collected after agents have done something, and thus most importantly, the last step after agents have done something.

If you want to collect before the first step (at time = 0), you can also add `self.datacollector.collect(self)` once in the init.

All example models already do this, this was the only test case that needed to be adjusted.

See for further details the _Implementation details_ section of the PR start.